python/pyspark/sql/tests/test_arrow.py

```diff
@@ -205,6 +205,49 @@ def test_toPandas_fallback_disabled(self):
                 with self.assertRaisesRegex(Exception, "Unsupported type"):
                     df.toPandas()
 
+    def test_toPandas_empty_df_arrow_enabled(self):
+        # SPARK-30537 test that toPandas() on an empty dataframe has the correct dtypes
+        # when arrow is enabled
+        from datetime import datetime, date
+        from decimal import Decimal
+
+        schema = StructType(
+            [
+                StructField("a", StringType(), True),
+                StructField("a", IntegerType(), True),
+                StructField("c", TimestampType(), True),
```

Comment:
We will probably also have to add `TimestampNTZ` and `DayTimeIntervalType` to test. `DayTimeIntervalType` might have an issue (see also https://github.com/apache/spark/pull/34631/files#r751947212)